pkg/azurefile/controllerserver.go

```diff
@@ -853,32 +853,39 @@ func (d *Driver) CreateSnapshot(ctx context.Context, req *csi.CreateSnapshotRequ
 		}, nil
 	}
 
-	shareURL, err := d.getShareURL(ctx, sourceVolumeID, req.GetSecrets())
-	if err != nil {
-		return nil, status.Errorf(codes.Internal, "failed to get share url with (%s): %v", sourceVolumeID, err)
-	}
+	// shareURL, err := d.getShareURL(ctx, sourceVolumeID, req.GetSecrets())
+	// if err != nil {
+	// 	return nil, status.Errorf(codes.Internal, "failed to get share url with (%s): %v", sourceVolumeID, err)
+	// }
+
+	// snapshotShare, err := shareURL.CreateSnapshot(ctx, azfile.Metadata{snapshotNameKey: snapshotName})
+	// if err != nil {
+	// 	return nil, status.Errorf(codes.Internal, "create snapshot from(%s) failed with %v, shareURL: %q", sourceVolumeID, err, shareURL)
+	// }
 
-	snapshotShare, err := shareURL.CreateSnapshot(ctx, azfile.Metadata{snapshotNameKey: snapshotName})
+	shareFileClient := d.cloud.FileClient.WithSubscriptionID(subsID)
+
+	snapshotShare, err := shareFileClient.CreateFileShare(ctx, rgName, accountName, &fileclient.ShareOptions{Name: snapshotName, RequestGiB: 1}, "snapshot")
 	if err != nil {
-		return nil, status.Errorf(codes.Internal, "create snapshot from(%s) failed with %v, shareURL: %q", sourceVolumeID, err, shareURL)
+		return nil, status.Errorf(codes.Internal, "create snapshot from(%s) failed with %v, accountName: %q", sourceVolumeID, err, accountName)
 	}
 
-	klog.V(2).Infof("Created share snapshot: %s", snapshotShare.Snapshot())
+	klog.V(2).Infof("Created share snapshot: %s", *snapshotShare.ID)
 
-	properties, err := shareURL.GetProperties(ctx)
+	// properties, err := shareFileClient.GetServiceProperties(ctx, rgName, accountName)
 	if err != nil {
-		return nil, status.Errorf(codes.Internal, "failed to get snapshot properties from (%s): %v", snapshotShare.Snapshot(), err)
+		return nil, status.Errorf(codes.Internal, "failed to get snapshot properties from (%s): %v", *snapshotShare.ID, err)
 	}
 
-	tp := timestamppb.New(properties.LastModified())
+	tp := timestamppb.New(snapshotShare.LastModifiedTime.Time)
```

Comment:
`LastModifiedTime` is a pointer, always check whether pointer is nil otherwise there could be panic